iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -87,11 +87,28 @@ PoshRuntime::PoshRuntime(cxx::optional<const ProcessName_t*> name, const bool do
     {
         LogWarn() << "Running applications on 32-bit architectures is not supported! Use at your own risk!";
     }
+
+    /// @todo #404 Use posix::FileLock here
+    //
+    // if (m_fileLock.hasError())
```

Comment:
Other ideas discussed in #404 included a ping pong between RouDi and the app or even some sort of protocol. IMHO a file lock solution is straight forward, no communication with RouDi is necessary. Resources like the UNIX domain socket or the message queue should not be cleaned up by RouDi, but instead by its owner.